services/core-api/app/api/verifiable_credentials/manager.py

```diff
@@ -195,3 +288,116 @@ def produce_map_01_credential_payload(cls, did: str, permit_amendment: PermitAme
+            geolocation=
+            f'https://plus.codes/{plus_code_encode(permit_amendment.mine.latitude, permit_amendment.mine.longitude)}',
+            verifiedByCAB=True)
+
+        products = [
+            cc.Product(
+                name=c,
+                classifications=cc.Classification(
+                    scheme="https://unstats.un.org/unsd/classifications/Econ/cpc",
+                    classifierName=c),
+                verifiedByCAB=False) for c in permit_amendment.mine.commodities
+        ]
+
+        untp_assessments = [
+            cc.ConformityAssessment(
+                referenceRegulation=cc.Regulation(
+                    id="https://www.bclaws.gov.bc.ca/civix/document/id/complete/statreg/96293_01",
+                    name="BC Mines Act",
+                    issuingBody=base.Party(
+                        name="BC Government",
+                        identifiers=[
+                            base.Identifier(identifierValue="did:indy:candy:LTNyw5R14J66CrF7tmV3i8")
+                        ]),
+                    effectiveDate=datetime(2024, 5, 14, tzinfo=pytz.timezone("UTC")).isoformat()),
+                subjectFacilities=[facility],
+                subjectProducts=products,
+                sustainabilityTopic=codes.SustainabilityTopic.Governance_Compliance)
+        ]
+        issue_date = permit_amendment.issue_date
+        issuance_date_str = datetime(
+            issue_date.year, issue_date.month, issue_date.day, 0, 0, 0,
+            tzinfo=pytz.timezone("UTC")).isoformat()
+
+        cred = cc.ConformityAttestation(
+            id="http://example.com/attestation/123",
+            assessmentLevel=codes.AssessmentAssuranceCode.GovtApproval,
+            type=codes.AttestationType.Certification,
+            description=
+            "This is a conformity attestation for the existence and good standing of a mining permit under the Mines Act within British Columbia (a province of Canada).",
+            scope=cc.ConformityAssessmentScheme(
+                id=
+                "https://github.com/bcgov/bc-vcpedia/blob/main/credentials/bc-mines-act-permit/1.1.1/governance.md",
+                name="BC Mines Act Permit Credential (1.1.1) Governance Documentation"),
+            issuedBy=untp_party_cpo,
+            issuedTo=untp_party_business,
+            validFrom=issuance_date_str,                                                                                                                                   #shouldn't this just be in the w3c wrapper
+            assessments=untp_assessments)
+
+        class W3CCred(BaseModel):
+            model_config = ConfigDict(
+                populate_by_name=True, json_encoders={datetime: lambda v: v.isoformat()})
+
+            context: List[Union[str, dict]] = Field(alias="@context")
+            type: List[str]
+            issuer: dict[str, str]
+            issuanceDate: str
+            credentialSubject: cc.ConformityAttestation
```

Comment:
Any reason for keeping the class inside the function? Thinking it might be good to move it out of there in case we want to reference it elsewhere

Reply:
no, but i should probably move that to the library (or find a library that already implements the w3c vc data model). 

https://www.w3.org/TR/vc-data-model-2.0/